app/models/mixins/custom_attribute_mixin.rb

```diff
@@ -27,6 +27,16 @@ module CustomAttributeMixin
     def self.custom_keys
       CustomAttribute.where(:resource_type => base_class).map(&:name).uniq
     end
+
+    def self.add_custom_attribute(custom_attribute)
+      virtual_column(custom_attribute.to_sym, :type => :string)
+
+      define_method(custom_attribute.to_sym) do
+        attr = custom_attribute.sub(CUSTOM_ATTRIBUTES_PREFIX, "")
+        value = miq_custom_attributes.or(ems_custom_attributes).find_by('lower(name) = ?', attr).try(:value)
```

Comment:
1. I would try `pluck(:value)` and see if that will work for you.
2. If not, use `select(:value)` since that is the only column you care about.
3. no reason for the variable `value`, just use `|| ''` on this line